src/xrCore/Crypto/xr_sha.h

```diff
@@ -1,22 +1,29 @@
 #pragma once
 #include "xrCore/xrCore.h"
 #include <cryptopp/sha.h>
+
+#include <optional>
 #include <array>
 
 namespace crypto
 {
+
+using sha_process_yielder = fastdelegate::FastDelegate1<long>;
```

Comment:
Yielder should probably take size_t as an argument.

Reply:
The whole code in Engine uses long. So this change has a large impact. I have tried to do this, but it should be done in the context of the separate task.